### PR TITLE
Stencil v3

### PR DESCRIPTION
**Pull request by GCHQ-Developer-530**
## Summary of the changes
Upgrade stencil to V3 and stencil react output target to 0.4.0, and updated types coming from
interface to come from utils/types to fix build errors from V3 upgrade

## Related issue
https://github.com/mi6/ic-ui-kit/issues/386

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [x] I have manually accessibility tested any changes, if relevant.
- [x] I have ensured any changes match the Figma component library.

### CODE REVIEW

**Comment by gd2910** [comment]
Looks like there is a lot of prettier/linting changes in the files, is this intentional?

File: packages/react/src/react-component-lib/createComponent.tsx
[the remark this answers is not in the document]
Comment (GCHQ-Developer-530) [reply packages/react/src/react-component-lib/createComponent.tsx:1]: It automatically made those changes itself when running npm build.

**Comment by GCHQ-Developer-530** [comment]
> Looks like there is a lot of prettier/linting changes in the files, is this intentional?

Yeah, most of them were made automatically when running npm build, but others were necessary as it wouldn't build without the prettier/linting updates

**Review by gd2910 — APPROVED** [review]

File: packages/react/src/react-component-lib/createComponent.tsx
[the remark this answers is not in the document]
Comment (GCHQ-Developer-847) [reply packages/react/src/react-component-lib/createComponent.tsx:1]: Ah ok great 👍

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Pull request merged**